### ISSUE TITLE
Early Stopping Problem for first training model (Basic feature extractor (Vanilla Conv2D))

### ISSUE THREAD

**Issue by Tabrez-dev** [AIWintermuteAI/Speech-to-Intent-Micro#5]
Im using ubuntu OS and i have used a virtual env to install all requirements and have used speech_to_intent_tf_keras_edited.ipynb jupyter notebook to run the cells.

![Screenshot from 2024-03-07 09-12-30](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/82bf3ec1-957d-48d2-842c-4f15e770805d)
This is the cell i am getting early stopping error.

![Screenshot from 2024-03-07 09-12-16](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/e66a0852-2f14-4685-a92c-244857456049)

The early stopping error occurs at epoch 11 but if you look closely from epoch 2 val_loss value does not improve at all. Is there any way to fix this?

I believe all previous cells have run successfully.
Here are the screen shots of previous outputs cells before training. 
![Screenshot from 2024-03-07 09-29-44](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/05679b3a-8044-4f8c-b66e-40a93962c5af)

![Screenshot from 2024-03-07 09-30-03](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/38e27da3-8479-4977-8f0a-94c88a38b247)

![Screenshot from 2024-03-07 09-30-11](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/aa2fd66a-ac13-4b43-849e-7df516130011)

![Screenshot from 2024-03-07 09-30-26](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/9ea412f1-5a1a-4162-86cc-aca26f7833d5)

![Screenshot from 2024-03-07 09-30-35](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/e8151533-d7a1-4aa8-b410-6786d1bfb446)

![Screenshot from 2024-03-07 09-31-17](https://github.com/AIWintermuteAI/Speech-to-Intent-Micro/assets/75200693/4908c093-825f-49c1-a37c-8b8a1753e9da)

**Comment by AIWintermuteAI** [comment]
Hello!
Yes, something looks strange with validation loss not decreasing at all, while training loss does decrease. I'll try reproducing it within this week.

**Comment by AIWintermuteAI** [comment]
Hello, @Tabrez-dev !
I identified the problem:
```
generate_data = True #change that to True the first time you are running the code
```
this line was set to False, which led to using old invalidated slots and intents for validation / test data.
It does say ` #change that to True the first time you are running the code` there, but I am guilty myself of running the code without looking at the comments at times...

I changed the value to True and made a couple of other tweaks. It should work now.

---------------
Did you know caffeine powers both code and coders? If my tweaks saved your day, consider fueling my next breakthrough with a coffee on BuyMeACoffee. Thanks a latte! ☕

https://www.buymeacoffee.com/hardwareai

**Comment by Tabrez-dev** [comment]
Since you say you have done some tweaks is it paossible to share the updated codes?

**Comment by Tabrez-dev** [comment]
Hi, @AIWintermuteAI i have seen the jupyter notebook. Thanks so much for helping out. What are some other tweaks did u do?

I have few questions can you please answer them in detail:

1. I am trying to implement this project on syntiant NDP120. If you have any idea on using the SDK and uploading the model on this kind of board.

If you haven't used the above mentioned board before can you list out general steps to upload the model generated into the hardware?

2. What is the point of other models after basic vanilla conv 2? Will they enhance the model created by basic vanilla conv 2 or produce their own model and at the end we select the best one out of all? I.e highest accuracy one?

3. Im thinking of running the model for 75 epochs for the basic conv 2d. What other parameters i can change to get better performance from the model? Thanks!

**Comment by AIWintermuteAI** [comment]
Just bug fixes, related to package API updates, you can check the diffs. Plus I made it easier to run on colab. Nothing that would change the performance.
1. Have not used that board personally. If it has ML accelerator, you need to convert the resulting .tflite model with their tooling.
2. The notebook is meant to be as series of experiments - as you see there is a cell after all the training is done, which calculates the accuracy and prints the best model.
3. You can play around with learning schedule, optimizer and possibly the number of kernels in layers. 

I'll close the issue as the original problem was solved.

---------------
Ever heard of the coder's fuel? Yep, it's coffee! If my tweaks saved your day, why not share a cup with me on BuyMeACoffee? Your support keeps the code flowing smoothly. Cheers! ☕

https://www.buymeacoffee.com/hardwareai
